core/src/main/scala/weaponregex/model/regextree/predefinedCharClassNode.scala

```diff
@@ -12,11 +12,22 @@ import weaponregex.model.Location
 case class PredefinedCharClass(charClass: String, override val location: Location)
     extends Leaf(charClass, location, """\""")
 
-/** POSIX character class leaf node
+/** Unicode character class leaf node
   * @param property
   *   The class character property
+  * @param propValue
+  *   The value of the [[property]]
   * @param location
   *   The [[weaponregex.model.Location]] of the node in the regex string
   */
-case class POSIXCharClass(property: String, override val location: Location, isPositive: Boolean = true)
-    extends Leaf(property, location, if (isPositive) """\p{""" else """\P{""", "}")
+case class UnicodeCharClass(
+    property: String,
+    override val location: Location,
+    isPositive: Boolean = true,
+    propValue: String = ""
+) extends Leaf(
+      if (propValue.isEmpty) property else "%s=%s".format(property, propValue),
```

Comment:
I think it's fine to use string interpolation here instead of `.format`. It's more in line with the rest of the project (and it's been so long I've used I actually had to look up if `format` was what I thought it was and not something special 😅)

Reply:
Updated! 🧐